Re vamp lobbyform issue#2

JonWong37 opened the pull request.

JonWong37 commented:
First screenshot is how the lobby currently looks.
Implemented the Button component into LobbyForm

Second screenshot shows hovering over the button high-lighting the first button.
Third screenshot shows hovering over the button and high-lighting the 2nd button.


<img width="1440" alt="Screen Shot 2022-12-01 at 7 30 15 PM" src="https://user-images.githubusercontent.com/110418682/205209848-996844a0-15ff-43ed-9108-5c433e0d3c56.png">

Second screenshot shows hovering over the button high-lighting the first button.
Third screenshot shows hovering over the button and high-lighting the 2nd button.
Added the hovering colors to the buttons. I set them up as 2 separate Css for each button.

<img width="1440" alt="Screen Shot 2022-12-01 at 7 30 33 PM" src="https://user-images.githubusercontent.com/110418682/205209924-a167a362-40c5-4485-88d1-a213cb54fbb5.png">
<img width="1440" alt="Screen Shot 2022-12-01 at 7 30 41 PM" src="https://user-images.githubusercontent.com/110418682/205209937-edc84f31-98d0-48d8-a5c3-f09fee6c443a.png">

Fourth shows onClick active for the button to change colors and shows responsiveness to when being clicked.

<img width="1440" alt="Screen Shot 2022-12-01 at 7 31 24 PM" src="https://user-images.githubusercontent.com/110418682/205209947-4616df25-a52f-41f9-b061-f8102d4af49a.png">

Fifth screenshot shows displays inputing target into the "Join lobby"

<img width="1440" alt="Screen Shot 2022-12-01 at 7 44 09 PM" src="https://user-images.githubusercontent.com/110418682/205210242-d6302c49-4797-4124-bfa0-de1e023dc3b5.png">

Potential Issues : The spacing for the button components for "Create Lobby" and "Join Lobby" are uneven. They're not perfectly aligned. I was manually setting the width of these components. There is definitely a better way to do it. 

The highlight within the button is done with setting the padding to 0 to get rid of div spacing issue that I previously mentioned. Please let me know on your feedback regarding this issue.

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -13,35 +15,52 @@ function LobbyForm({ socket }: Props): React.ReactElement {
     <div className={css(hostStyles.form)}>
       <Host socket={socket} />
-      <div className={css(hostStyles.or)}>- or -</div>
+      <div className={css(hostStyles.or)}></div>
```
is this div still necessary?

JonWong37 replied:
Currently left this div in to create spacing between the two components. I will look into a better way to create that space in a little bit.

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -56,16 +75,27 @@ function Join({ socket }: HostProps): ReactElement {
       <div>
-        <button onClick={onJoinLobby} className={css(hostStyles.button)}>
-          &gt;
+        <button
+          onClick={onJoinLobby}
+          className={css(hostStyles.button, hostStyles.button2)}
+        >
```
why do we need this button? shouldn't the Button component replace this completely?

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -13,35 +15,52 @@ function LobbyForm({ socket }: Props): React.ReactElement {
+        />
+      </div>
       <div>
         <button onClick={onHost} className={css(hostStyles.button)}>
```
Do we need this button? Shouldn't the Button component replace this completely?

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -13,35 +15,52 @@ function LobbyForm({ socket }: Props): React.ReactElement {
 
 type HostProps = Readonly<{
   socket: Socket;
+  onClick?: any;
```
avoid the `any` type. you should be able to type it with something like this

```
type SomeType = Readonly<{
  someFunction: (props: number) => void,
  someFunctionReturnsString: () => string,
}>
```

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -13,35 +15,52 @@ function LobbyForm({ socket }: Props): React.ReactElement {
 type HostProps = Readonly<{
   socket: Socket;
+  onClick?: any;
+  onChange?: any;
```
I don't think this is being used by the Host component

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -13,35 +15,52 @@ function LobbyForm({ socket }: Props): React.ReactElement {
   return (
     <div className={css(hostStyles.row)}>
-      <div className={css(hostStyles.text)}>Create a new game</div>
+      <div>
```
is this div around the Button component necessary?

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -79,26 +109,36 @@ const hostStyles = StyleSheet.create({
     minWidth: "32px",
-    backgroundColor: "#EEEEEE",
+    backgroundColor: "#1D1E33",
+    ":hover": {
+      backgroundColor: "#069AB4",
+    },
```
did you have specs for the hover & active states? If not, let's avoid it for now, can just leave it without the hover & active states.

Reasoning:
 If we're using components that are supposed to be like standard components, it would make sense that they would have the feature on them instead.

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -79,26 +109,36 @@ const hostStyles = StyleSheet.create({
+  },
+  button2: {
     ":hover": {
-      backgroundColor: "#DDDDDD",
+      backgroundColor: "#7139E0",
     },
     ":active": {
-      backgroundColor: "#CCCCCC",
+      backgroundColor: "#BA9CFC",
```
Let's use a better name than "button2", something like "actionButton" or like "joinButton" or something would be better than "button2"

Reasoning: 
If I'm new to the codebase, it's a lot easier to read "joinButton" rather than "button2" since "joinButton" gives context that it's for the "join" feature.

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -79,26 +109,36 @@ const hostStyles = StyleSheet.create({
+    border: "none",
     borderRadius: "8px 0 0 8px",
-    "::placeholder": {
+    "input::placeholder": {
```
can we just remove the placeholder bits? We shouldn't need to style the input as it should be already styled from the component itself (TextInput)

HowiJ commented on src/common/TextInput.tsx:
```diff
@@ -68,6 +68,10 @@ const styles = StyleSheet.create({
     background: "transparent",
     borderRadius: "60px",
     alignSelf: "flex-start",
+    "::placeholder": {
+      color: "#FFFFFF",
+      textAlign: "center",
+    },
```
This is the correct place to have the styling as opposed to the above comment. 

Reasoning: This is supposed to be a core component, if it should be a certain way, then it should be on the core component. Reduce random extra code

JonWong37 commented:
Moved the Host and Join components to their own files to reduce clogging on lobbyform.tsx
Rewrote the props to avoid using `any` type.
Removed unnecessary  <button> tags and <div> tags.

Removed all active and hover states currently in the CSS.
Reduced random lines in the CSS that seem to have no effect on the overall product.

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,68 @@
+import Button from "./common/Button";
+import TextInput from "./common/TextInput";
+
+type JoinProps = Readonly<{
+  socket: Socket;
+  onClick: () => void;
+  onChange: () => void;
```
take a look at your component definition, you're expecting 3 props: socket, onclick, and onchange but you're only using socket. Don't add unnecessary props

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,68 @@
+import Button from "./common/Button";
+import TextInput from "./common/TextInput";
+
+type JoinProps = Readonly<{
```
Change this to just Props instead of JoinProps

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,68 @@
+
+  return (
+    <div className={css(hostStyles.row)}>
+      <div>
```
do we need this div?

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,68 @@
+          value={lobbyID}
+        />
+      </div>
+      <div>
```
do we need this div?

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,68 @@
+}
+
+const hostStyles = StyleSheet.create({
+  button: {
```
I don't think this styling is used, remove if so

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -1,9 +1,10 @@
-import type { FormEvent, ReactElement } from "react";
+// import type { ReactElement } from "react";
```
remove this comment please

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -1,9 +1,10 @@
 import React from "react";
 import { StyleSheet, css } from "aphrodite";
-import useFormActions from "useFormActions";
+import Host from "./Host";
+import Join from "./Join";
```
nice

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -12,96 +13,16 @@ type Props = Readonly<{
-          &gt;
-        </button>
-      </div>
+      <Host socket={socket} onClick={() => {}} />
```
This onclick is just an empty function, seems kinda pointless, let's remove this unless there's a specific reason you need to pass an empty onClick to Host

HowiJ commented on src/Host.tsx:
```diff
@@ -0,0 +1,57 @@
+
+type HostProps = Readonly<{
+  socket: Socket;
+  onClick: () => void;
```
Not used, remove

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -12,96 +13,16 @@ type Props = Readonly<{
 
 const hostStyles = StyleSheet.create({
   or: {
-    textAlign: "center",
-    padding: "8px",
-    fontSize: "8px",
-  },
-  button: {
-    height: "100%",
-    border: "none",
-    borderRadius: "0 8px 8px 0",
-    minWidth: "32px",
-    backgroundColor: "#EEEEEE",
-    ":hover": {
-      backgroundColor: "#DDDDDD",
-    },
-    ":active": {
-      backgroundColor: "#CCCCCC",
-    },
-  },
-  input: {
-    padding: "8px",
-    fontSize: "12px",
-    minWidth: "160px",
-    fontFamily: "Arial",
-    border: "1px solid #EEEEEE",
-    borderRadius: "8px 0 0 8px",
-    "::placeholder": {
-      fontSize: "12px",
-      fontFamily: "Arial",
-    },
```
Nice cleanup

HowiJ reviewed (changes requested):
Looks good, but there's a lot of unused things. Keep an eye out for unused things and typically, if it's unused, remove it unless there's a comment specifically saying why it's unused

JonWong37 commented:
Uploaded new PR with issues resolved.

Removed unused code and empty divs.

Added a gap between the Host and Join components.

Changed names from HostProps and JoinProps to just props.

HowiJ commented on src/Host.tsx:
```diff
@@ -0,0 +1,45 @@
+  );
+}
+
+const hostStyles = StyleSheet.create({
```
change to 'styles'

HowiJ commented on src/Join.tsx:
```diff
@@ -0,0 +1,53 @@
+  );
+}
+
+const hostStyles = StyleSheet.create({
```
change to 'styles'

HowiJ commented on src/LobbyForm.tsx:
```diff
@@ -1,131 +1,32 @@
-      fontFamily: "Arial",
-    },
-  },
+const Styles = StyleSheet.create({
```
lowercase styles

reasoning: variable naming conventions

HowiJ reviewed (approved):
lgtm
